### PR TITLE
cached build for CI #496

### PR DESCRIPTION
**Pull request by archibate**

### CODE REVIEW

**Comment by archibate** [comment]
It just succeed? I thought need cmake check somehow.

**Comment by yuanming-hu** [comment]
Thanks! One thing to consider: `CMakeLists.txt` uses file globs to gather source files:
https://github.com/taichi-dev/taichi/blob/42604e8e1928e4281515324d9a7067285108a594/cmake/TaichiCore.cmake#L5

which means, for example, if a new `cpp` file is added, the `build` folder should be updated while `CMakeLists.txt` is modified. Does the caching mechanism work correctly in this case?

**Comment by archibate** [comment]
Should can, since `cmake` is executed every time we build. It'll re-glob and see what to change in `Makefile`.
https://github.com/taichi-dev/taichi/blob/a3a56ae9987c5f7465c2e9eabc88c4dcbd7914f3/appveyor.yml#L54

File: .travis.yml [hunk 1]
```diff
@@ -9,6 +9,18 @@ matrix:
...
       - MATRIX_EVAL="CC=clang && CXX=clang++ && PYTHON=python3.7"
 
+skip_commits:
+  files:
+    - README.md
+    - docs
+    - benchmarks
+    - examples
+    - misc
+    - '.*'
```
Comment (archibate) [inline .travis.yml]: Surprising, `.travis.yml` is not included, so smart!

**Comment by yuanming-hu** [comment]
> Should can, since cmake is executed every time we build. It'll re-glob and see what to change in Makefile.

**Comment by archibate** [comment]
Let's have a try! We can test it by adding `taichi/congratulation.cpp` that purposely makes an pre-compile error.

**Comment by archibate** [comment]
```
[ 20%] Building CXX object CMakeFiles/taichi_core.dir/taichi/congratulation.cpp.o

/Users/travis/build/taichi-dev/taichi/taichi/congratulation.cpp:1:2: error: "Congratulation! CI did recognized me!"

#error "Congratulation! CI did recognized me!"

 ^

1 error generated.
```
Travis build failed, in other words: we succeed!!

**Comment by yuanming-hu** [comment]
Cool! Now I understand the caching mechanism better. Thanks!

**Pull request merged**